arkouda/numeric.py

```diff
@@ -584,55 +584,29 @@ def arctan2(num: Union[numeric_scalars, pdarray], denom: Union[numeric_scalars,
     TypeError
         Raised if the parameter is not a pdarray
     """
-    if isinstance(num, pdarray) and isinstance(denom, pdarray):
-        return create_pdarray(
-            type_cast(
-                str,
-                generic_msg(
-                    cmd="efunc2vv",
-                    args={
-                        "func": "arctan2",
-                        "num": num,
-                        "denom": denom,
-                    },
-                ),
-            )
-        )
-    elif isinstance(num, pdarray) and isSupportedNumber(denom):
-        return create_pdarray(
-            type_cast(
-                str,
-                generic_msg(
-                    cmd="efunc2vs",
-                    args={
-                        "func": "arctan2",
-                        "num": num,
-                        "scalar": denom,
-                        "dtype": resolve_scalar_dtype(denom),
-                    },
-                ),
-            )
-        )
-    elif isSupportedNumber(num) and isinstance(denom, pdarray):
-        return create_pdarray(
-            type_cast(
-                str,
-                generic_msg(
-                    cmd="efunc2sv",
-                    args={
-                        "func": "arctan2",
-                        "scalar": num,
-                        "denom": denom,
-                        "dtype": resolve_scalar_dtype(num),
-                    },
-                ),
-            )
+    if not all(isSupportedNumber(arg) or isinstance(arg, pdarray) for arg in [num, denom]):
+        raise TypeError(
+            f"Unsupported types {type(num)} and/or {type(denom)}. Supported "
+            "types are numeric scalars and pdarrays. At least one argument must be a pdarray."
```

Comment:
I would expect this to be handled by the type checking. Is there something causing it to not handle this as expected?

Reply:
yeah i was surprised by this too, but the inclusion of `numeric_scalars` in the type hint make it so it doesn't do any type checking (at least from what I saw). I was thinking we would want to take a look at similar places where we do type hints with groups that aren't classes and see how we want to handle them. I can make an issue to dig into that if u agree with that way forward

Reply:
I think investigating this under a separate issue makes sense.

Reply:
Captured in #2528